pkg/function/checkRepository.go

```diff
@@ -55,22 +54,28 @@ func CheckRepository(ctx context.Context, cli kubernetes.Interface, tp param.Tem
 		PodOverride:  podOverride,
 	}
 	pr := kube.NewPodRunner(cli, options)
-	podFunc := CheckRepositoryPodFunc(cli, tp, namespace, encryptionKey, targetPaths)
-	return pr.Run(ctx, podFunc)
+	podFunc := CheckRepositoryPodFunc(cli, tp, encryptionKey, targetPaths)
+	return pr.RunEx(ctx, podFunc)
 }
 
-func CheckRepositoryPodFunc(cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey, targetPath string) func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
-	return func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+func CheckRepositoryPodFunc(cli kubernetes.Interface, tp param.TemplateParams, encryptionKey, targetPath string) func(ctx context.Context, pc kube.PodController) (map[string]interface{}, error) {
```

Comment:
```suggestion
func CheckRepositoryPodFunc(
	cli kubernetes.Interface,
	tp param.TemplateParams,
	encryptionKey,
	targetPath string,
) func(ctx context.Context, pc kube.PodController) (map[string]interface{}, error) {
```
Similar formatting change as above